### DISCUSSION TITLE
Cache for index-format-hook values

### DISCUSSION THREAD

**Discussion by vt-alt** [neomutt/neomutt#4520] [Development]
When `index-format-hook` is used not extensively (even for 5 lines with `~b` for a local mbox) - moving cursor becomes considerably slower and lagging. Perhaps regexps are re-executed on every movement, even though index is already displayed with relevant info.

Maybe it's possible to add some caching to `index-format-hook` values and not re-execute it on every movement? 

Example how I tried to use it:
```
index-format-hook resol "~b^fixes:"          Fixes
index-format-hook resol "~b^signed-off-by:"  PATCH
index-format-hook resol "~b^acked-by:"       Ack
index-format-hook resol "~b^reviewed-by:"    Review
index-format-hook resol "~b^tested-by:"      Tested
set index_format="%4C %Z %[%y %b %d %H:%M] %-15.15n %-6@resol@ (%4l%?X?+%X&?) %H %s"
```
This is to show patch status on [lore](https://lore.kernel.org/)-style mailboxes.

ps. This seem similar to #3510 but reporter closed the issue.

**Comment by gahr** [comment]
Do you have an idea how we might "cache" these results? The matching pattern might depend on the email and the environment, so the mere passing of time could cause a different pattern to match. See the canonical example over at our [reference](https://neomutt.org/guide/configuration.html#22-%C2%A0dynamically-changing-index_format-using-patterns):
```
set index_format="%4C %-6@date@ %-15.15F %Z (%4c) %s"

index-format-hook  date  "~d<1d"    "%[%H:%M]"
index-format-hook  date  "~d<1m"    "%[%a %d]"
index-format-hook  date  "~d<1y"    "%[%b %d]"
index-format-hook  date  "~A"       "%[%m/%y]"
``` 

If you wait long enough and change your selection, emails that previously matched `~d<1d` might now match `~d<1m`.
